### ISSUE TITLE
Magnet link not responding

### ISSUE THREAD

**Issue by b1fly** [Sam-Max/Rclone-Tg-Bot#19]
/mirror
/qbmirror

Nothing happens, even though I reply to the magnet link, there is still no response

**Comment by Sam-Max** [comment]
It is working fine, check again what you did, and /mirror is not for torrent only /qbmirror

**Comment by b1fly** [comment]
> It is working fine, check again what you did and if the bot is running, and /mirror is not for torrent only /qbmirror

The console doesn't respond with any logs, it's hard for me to check what's wrong

**Comment by Sam-Max** [comment]
> 

If bot is working use /logs commands

**Comment by b1fly** [comment]
> > 
> 
> If bot is working use /logs command

![image](https://user-images.githubusercontent.com/108970034/181906956-bb160715-dd96-41ac-90ed-e971def6c38d.png)

**Comment by b1fly** [comment]
![image](https://user-images.githubusercontent.com/108970034/181908491-47c5a0c3-e05a-4129-948c-6106a095183f.png)


`2022-07-30 11:12:04,502 INFO MainThread telethon.client.uploads Uploading file of 3805 bytes in 1 chunks of 131072
2022-07-30 11:15:52,661 INFO MainThread bot _get_hash_magnet
2022-07-30 11:15:52,663 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6baf0>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,668 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe126c6d090>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,671 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6f640>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v2/auth/login
2022-07-30 11:15:52,674 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c69960>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,677 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe126c6df90>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,679 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6ec50>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v2/auth/login
2022-07-30 11:15:52,780 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6fc40>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,783 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe126c6d810>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,786 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6ae00>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v2/auth/login
2022-07-30 11:15:52,788 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6db10>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,791 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fe126c6cd60>: Failed to establish a new connection: [Errno 111] Connection refused')': /
2022-07-30 11:15:52,793 WARNING MainThread urllib3.connectionpool Retrying (Retry(total=0, connect=0, read=1, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6f910>: Failed to establish a new connection: [Errno 111] Connection refused')': /api/v2/auth/login
2022-07-30 11:15:52,794 ERROR MainThread pyrogram.dispatcher Failed to connect to qBittorrent. Connection Error: ConnectionError(MaxRetryError("HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6c1c0>: Failed to establish a new connection: [Errno 111] Connection refused'))"))
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/usr/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fe126c6e5c0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 815, in urlopen
    return self.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6e5c0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 378, in _request_manager
    return self._request(**kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 428, in _request
    response = self._session.request(
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 538, in request
    return super(QbittorrentSession, self).request(method, url, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 565, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6e5c0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/app/bot/downloaders/qbit/qbit_downloader.py", line 47, in add_qb_torrent
    tor_info = self.client.torrents_info(torrent_hashes=self.ext_hash)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/decorators.py", line 119, in wrapper
    return func(client, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/decorators.py", line 164, in wrapper
    response = func(client, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/decorators.py", line 90, in wrapper
    client.auth_log_in(**get_requests_kwargs(**kwargs))
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/auth.py", line 96, in auth_log_in
    self._post(_name=APINames.Authorization, _method="login", data=creds, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 327, in _post
    return self._request_manager(
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 388, in _request_manager
    raise APIConnectionError(error_message, response=response)
qbittorrentapi.exceptions.APIConnectionError: Failed to connect to qBittorrent. Connection Error: ConnectionError(MaxRetryError("HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6e5c0>: Failed to establish a new connection: [Errno 111] Connection refused'))"))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/connection.py", line 95, in create_connection
    raise err
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/connection.py", line 85, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/usr/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 205, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fe126c6c1c0>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 815, in urlopen
    return self.urlopen(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.10/dist-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6c1c0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 378, in _request_manager
    return self._request(**kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 428, in _request
    response = self._session.request(
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 538, in request
    return super(QbittorrentSession, self).request(method, url, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/requests/adapters.py", line 565, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6c1c0>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/pyrogram/dispatcher.py", line 240, in handler_worker
    await handler.callback(self.client, *args)
  File "/usr/src/app/bot/core/handlers/handle_mirror_cm.py", line 29, in handle_qbit_mirror_command
    await mirror(client, message, isQbit=True)
  File "/usr/src/app/bot/core/handlers/handle_mirror_cm.py", line 114, in mirror
    state, message, path, name = await qbit_dl.add_qb_torrent(link)
  File "/usr/src/app/bot/downloaders/qbit/qbit_downloader.py", line 90, in add_qb_torrent
    self.client.auth_log_out()
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/decorators.py", line 90, in wrapper
    client.auth_log_in(**get_requests_kwargs(**kwargs))
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/auth.py", line 96, in auth_log_in
    self._post(_name=APINames.Authorization, _method="login", data=creds, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 327, in _post
    return self._request_manager(
  File "/usr/local/lib/python3.10/dist-packages/qbittorrentapi/request.py", line 388, in _request_manager
    raise APIConnectionError(error_message, response=response)
qbittorrentapi.exceptions.APIConnectionError: Failed to connect to qBittorrent. Connection Error: ConnectionError(MaxRetryError("HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /api/v2/auth/login (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe126c6c1c0>: Failed to establish a new connection: [Errno 111] Connection refused'))"))
2022-07-30 11:16:52,871 INFO MainThread telethon.client.uploads Uploading file of 20148 bytes in 1 chunks of 131072
`

**Comment by b1fly** [comment]
I tried to deploy it again with arm64 vps, the torrent download and the direct link download were normal, and there was no error, but the file was automatically uploaded to the cloud after downloading, and the upload path location will not be prompted. I need to look for them one by one in my onedrive, I thought it would be uploaded to the root directory, and when I downloaded the resources to the local, I saw that the path was INFO MainThread bot Cleaning Download: /usr/src/app/Downloads/16592395300020235/cmder.7z
![image](https://user-images.githubusercontent.com/108970034/182009487-7d95e528-d8de-4fbc-8c9d-2fa019810cc0.png)
![image](https://user-images.githubusercontent.com/108970034/182009496-5e525f91-3cd4-4db9-98ae-4ed4160f6d36.png)


I actually searched for this directory /usr/src/app/Downloads/ and found nothing.
I hope that you can add options in config.env to customize the download directory. You can change the configuration options of aria2 and qbittorrent, such as: port, address, account number, password, custom modification, which can increase security risks, and the cloud can add custom upload to A certain directory, so it won't be cluttered

This is just my suggestion, your bot open source is great.

**Comment by b1fly** [comment]
I think I should add an option to clean up download interruptions and download failed files, because I just accidentally mirrored all the files on onedrive to telegram, my vps instantly evaporated 8 GB of hard disk space, and I can't find where the files are on the vps. , causing me to be unable to clean up.

**Comment by b1fly** [comment]
I do not know why? I redeployed and the download magnet keeps failing
![image](https://user-images.githubusercontent.com/108970034/182019824-b213f073-874a-40ff-9fbb-13de847f7a22.png)
![image](https://user-images.githubusercontent.com/108970034/182019899-4d916f70-cf87-4105-950d-c8e7c79f5f07.png)

**Comment by Sam-Max** [comment]
> I will add this soon

**Comment by Sam-Max** [comment]
> I think I should add an option to clean up download interruptions and download failed files, because I just accidentally mirrored all the files on onedrive to telegram, my vps instantly evaporated 8 GB of hard disk space, and I can't find where the files are on the vps. , causing me to be unable to clean up.

I will add this soon

**Comment by Sam-Max** [comment]
> 

Added options in config.env to customize the download directory.

**Comment by b1fly** [comment]
>
> > 
> 
> Added options in config.env to customize the download directory.

Sir, this is great. I've been using it for a few days and I found that it gets stuck if I mirror multiple files to the cloud.

Personal request, I think a button similar to the current task status should be added. When there are many tasks, I need to keep searching in the robot for a few tasks that I have added, which is very troublesome. It is better to add a button to display which tasks are currently in progress. , which will save time
